backend/database.py

```
@@ -345,6 +394,24 @@ def clear_files():
+    """
+    Return {path: (size, last_modified)} for every indexed file.
+
+    Used by incremental re-indexing to decide which files are unchanged
+    since the previous index build.
+    """
+    conn = get_connection()
+    try:
+        cursor = conn.cursor()
+        cursor.execute('SELECT path, size, last_modified FROM files')
+        return {row['path']: (row['size'], row['last_modified']) for row in cursor.fetchall()}
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)